task_manager/task_manager.cc

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "task_manager.hh"
+#include "db/config.hh"
+
+seastar::gate& task_manager::async_gate() {
+    return _gate;
+}
+
+abort_source& task_manager::abort_source() {
+    return _as;
+}
+
+std::chrono::seconds task_manager::task_ttl() const {
+    return std::chrono::seconds(_cfg.task_ttl_in_seconds());
+}
+
+task_manager::tasks_per_module& task_manager::get_tasks_per_module() {
+    return _tasks_per_module;
+}
+
+task_manager::task_map& task_manager::get_all_tasks() {
+    return _all_tasks;
+}
+
+task_manager::task_ptr task_manager::get_task_this_shard(task_id id) const {
+    auto it = _all_tasks.find(id);
+    if (it != _all_tasks.end()) {
+        return it->second;
+    }
+    return nullptr;
+}
+
+std::vector<std::string> task_manager::get_modules() const {
+    return {"repair", "compaction"}; // TODO Deexie change to _modules
+}
+
+void task_manager::register_task(task_ptr task) {
+    auto id = task->get_status().id;
+    _tasks_per_module[task->get_module()].insert({id, task});
+    _all_tasks[id] = task;
```

Comment:
I think that eventually this would be done via the `task_manager::module` which can be used to create the task and register it.

The way I see it, the module class should embed an `std::unordered_map<task_id, task_ptr>`
so the task_manager will not have a map of maps like tasks_per_module.

Instead, it should just have a map of modules and access the per-module data using the module object.